Druid-Tests/DRSCCPConstantFoldingTest.class.st

```diff
@@ -219,6 +219,41 @@ DRSCCPConstantFoldingTest >> testConstantFoldingFalseOrRegisterIsRegister [
 	self assertLastInstructionOf: cfg isCopyOf: copy operand1.
 ]
 
+{ #category : #arithmetic }
+DRSCCPConstantFoldingTest >> testConstantFoldingLoadConstant [
+
+	| cfg |
+	cfg := self testConstantFoldingOf: DRLoad between: 8 asDRValue and: 9999 asDRValue.
+
+	self assert: cfg instructions allButLast last isCopy.
+	self
+		assert: cfg instructions allButLast last operand1 expression
+		equals: '(coInterpreter int64AtPointer:  9999)'
```

Comment:
Nice! Now what would be important for a next iteration is to decide what is *constant at JIT compile time* and what it's not. Because if the value at address 9999 never changes it is constant and we can load it at compile time. Otherwise, we need to defer that to runtime.